framerail/src/routes/[slug]/[...extra]/page.svelte

```diff
@@ -0,0 +1,339 @@
+<script lang="ts">
+  export let data
+  import { goto, invalidateAll } from "$app/navigation"
+  import { onMount } from "svelte"
+
+  let showMoveAction = false
+  let showHistory = false
+  let moveInputNewSlugElem: HTMLInputElement
+  let revisionList = []
+
+  async function handleDelete() {
+    let fdata = new FormData()
+    fdata.set("site-id", data.site.siteId)
+    fdata.set("page-id", data.page.pageId)
+    await fetch(`/${data.page.slug}`, {
+      method: "DELETE",
+      body: fdata
+    })
+    invalidateAll()
+  }
+
+  function navigateEdit() {
+    let options: string[] = []
+    if (data.options.noRender) options.push("norender")
+    options = options.map((opt) => `/${opt}`)
+    goto(`/${data.page.slug}${options.join("")}/edit`, {
+      noScroll: true
+    })
+  }
+
+  function cancelEdit() {
+    let options: string[] = []
+    if (data.options.noRender) options.push("norender")
+    options = options.map((opt) => `/${opt}`)
+    goto(`/${data.page.slug}${options.join("")}`, {
+      noScroll: true
+    })
+  }
+
+  async function saveEdit() {
+    let form = document.getElementById("editor")
+    let fdata = new FormData(form)
+    fdata.set("site-id", data.site.siteId)
+    fdata.set("page-id", data.page.pageId)
+    await fetch(`/${data.page.slug}/edit`, {
+      method: "POST",
+      body: fdata
+    })
+    goto(`/${data.page.slug}`, {
+      noScroll: true
+    })
+  }
+
+  async function handleMove() {
+    let form = document.getElementById("page-move")
+    let fdata = new FormData(form)
+    let newSlug = fdata.get("new-slug")
+    if (!newSlug) {
+      moveInputNewSlugElem.style.outline = "1px solid red"
+      return
+    } else {
+      moveInputNewSlugElem.style.outline = ""
+    }
+    fdata.set("site-id", data.site.siteId)
+    fdata.set("page-id", data.page.pageId)
+    await fetch(`/${data.page.slug}/move`, {
+      method: "POST",
+      body: fdata
+    })
+    goto(`/${newSlug}`, {
+      noScroll: true
+    })
+    showMoveAction = false
+  }
+
+  async function handleHistory() {
+    let fdata = new FormData()
+    fdata.set("site-id", data.site.siteId)
+    fdata.set("page-id", data.page.pageId)
+    revisionList = await fetch(`/${data.page.slug}/history`, {
+      method: "POST",
+      body: fdata
+    }).then((res) => res.json())
+    showHistory = true
+  }
+
+  onMount(() => {
+    if (data?.options.history) handleHistory()
+  })
+</script>
+
+<h1>UNTRANSLATED:Loaded page</h1>
+<p class="spin-yay">
+  UNTRANSLATED:This is a generic page renderer loaded as a component.
+</p>
+<p>
+  UNTRANSLATED:Response <textarea class="debug">{JSON.stringify(data, null, 2)}</textarea>
+</p>
+
+<h2>{data.pageRevision.title}</h2>
+
+<hr />
+
+<div class="page-content">
+  {#if data.options?.noRender}
+    UNTRANSLATED: Content not shown.
```

Comment:
This is fine for now. Eventually we should show a message and the page source underneath.